### PR TITLE
D3ASIM-3188: Enhancing dict key validator and value type.

### PR DESCRIPTION
**Pull request by faizan2590**

### CODE REVIEW

File: d3a_interface/utils.py [hunk 1]
```diff
@@ -69,6 +69,14 @@ def key_in_dict_and_not_none_and_not_str_type(d, key):
     return key_in_dict_and_not_none(d, key) and not isinstance(d[key], str)
 
 
+def key_in_dict_and_not_none_and_list_type(d, key):
+    return key_in_dict_and_not_none(d, key) and not isinstance(d[key], list)
+
```
Comment (Hassan754) [inline d3a_interface/utils.py:74]: In several past commits we needed this function , thanks for adding it

**Review by Hassan754 — APPROVED** [review]

File: d3a_interface/utils.py [hunk 2]
```diff
@@ -69,6 +69,14 @@ def key_in_dict_and_not_none_and_not_str_type(d, key):
     return key_in_dict_and_not_none(d, key) and not isinstance(d[key], str)
 
 
+def key_in_dict_and_not_none_and_list_type(d, key):
+    return key_in_dict_and_not_none(d, key) and not isinstance(d[key], list)
```
Comment (hannesdiedrich) [inline d3a_interface/utils.py:73]: do you mean
```suggestion
    return key_in_dict_and_not_none(d, key) and isinstance(d[key], list)
```

File: d3a_interface/utils.py [hunk 3]
```diff
@@ -69,6 +69,14 @@ def key_in_dict_and_not_none_and_not_str_type(d, key):
...
+
+
+def key_in_dict_and_not_none_and_dict_type(d, key):
+    return key_in_dict_and_not_none(d, key) and not isinstance(d[key], dict)
```
Comment (hannesdiedrich) [inline d3a_interface/utils.py:77]:
```suggestion
    return key_in_dict_and_not_none(d, key) and isinstance(d[key], dict)
```

**Comment by fievelk** [comment]
Where is this used? In general, I'm in favor of modularizing code into functions, as long as the function is used often. If not, a one-liner might just be as good (and saves some time).

**Comment by faizan2590** [comment]
> Where is this used? In general, I'm in favor of modularizing code into functions, as long as the function is used often. If not, a one-liner might just be as good (and saves some time).

Planning to use it [here](https://github.com/gridsingularity/d3a-web/pull/1135#discussion_r558485042) and its good to modularise this logic as needed quite often

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by Hassan754 — APPROVED** [review]

**Pull request merged**